src/LibraryViewExtension/LibraryViewController.cs

```diff
@@ -149,6 +152,18 @@ public void ImportLibrary()
                 dynamoViewModel.ImportLibraryCommand.Execute(null)
             ));
         }
+        /// <summary>
+        /// This function adds the events to instrumentation
+        /// </summary>
+        /// <param name="eventName"> Event Name</param>
+        /// <param name="data">Data</param>
+        public void AddInstrumentation(string eventName, string data)
+        {
+            if (eventName == "Search" || eventName == "Filter-Categories" || eventName == "Search-NodeAdded")
```

Comment:
do you do this to try to prevent others from logging whatever they want via the library?

Reply:
yes. I will add more comments.